src/main/java/pro/belbix/ethparser/web3/deployer/parser/DeployerTransactionsParser.java

```diff
@@ -0,0 +1,195 @@
+    private final DeployerDbService deployerDbService;
+    private final EthBlockService ethBlockService;
+    private final ParserInfo parserInfo;
+    private long parsedTxCount = 0;
+    private Instant lastTx = Instant.now();
+
+    public DeployerTransactionsParser(Web3Service web3Service, DeployerDecoder deployerDecoder,
+                                      DeployerDbService deployerDbService, EthBlockService ethBlockService,
+                                      ParserInfo parserInfo)
+    {
+        this.web3Service = web3Service;
+        this.deployerDecoder = deployerDecoder;
+        this.deployerDbService = deployerDbService;
+        this.ethBlockService = ethBlockService;
+        this.parserInfo = parserInfo;
+    }
+
+    public void startParse()
+    {
+        log.info("Start parse Deployer transactions");
+        parserInfo.addParser(this);
+        web3Service.subscribeOnTransactions(transactions);
+        new Thread(() ->
+        {
+            while (run.get())
+            {
+                Transaction transaction = null;
+                try
+                {
+                    transaction = transactions.poll(1, TimeUnit.SECONDS);
+                }
+                catch (InterruptedException ignored)
+                {
+                }
+                DeployerDTO dto = parseDeployerTransaction(transaction);
+                if (dto != null)
+                {
+                    lastTx = Instant.now();
+                    try
+                    {
+                        boolean success = deployerDbService.save(dto);
+                        if (success)
+                        {
+                            output.put(dto);
+                        }
+                    }
+                    catch (Exception e)
+                    {
+                        log.error("Can't save " + dto.toString(), e);
+                    }
+                }
+            }
+        }).start();
+    }
+
+    public DeployerDTO parseDeployerTransaction(Transaction tx)
+    {
+        DeployerDTO deployerDTO = null;
+        if (tx != null)
+        {
+            incrementAndPrintCount(tx);
+            DeployerTx deployerTx = decodeTransaction(tx);
+            if (deployerTx != null)
+            {
+                deployerDTO = deployerTx.toDto();
+                deployerDTO.setBlockDate(ethBlockService.getTimestampSecForBlock(tx.getBlockHash(),
+                        tx.getBlockNumber().longValue()));
+                print(deployerDTO);
+            }
+        }
+        return deployerDTO;
+    }
+
+    private void incrementAndPrintCount(Transaction tx)
+    {
+        parsedTxCount++;
+        if (parsedTxCount % LOG_LAST_PARSED_COUNT == 0)
+        {
+            log.info("Deployer parsed " + parsedTxCount + ", last block: " + tx.getBlockNumber());
+        }
+    }
+
+    private DeployerTx decodeTransaction(Transaction tx)
+    {
```

Comment:
better check tx at the start of the method and return null
if(!isValid(tx)) {
return null;
}

Reply:
avoid layers with if statements
simplified it
as example
if(...) {
 return ...
}

Reply:
why the decode method in the parser? move it to the decoder

Reply:
I had it in parser because that's where it is for uniswapTransactionParser class. I agree it should move though and I moved it.